### DISCUSSION TITLE
Manage the steps manually on the initializing process

### DISCUSSION THREAD

**Discussion by lucasmp2001** [hakimio/ngx-ui-tour#65] [Q&A] [answered]
### Hello guys!

I have one question.. since we can manipulate the steps using the custom template, and also order the steps manually too, it's possible to start on a custom step, and stop the tour on a specific step, doing this on the initializing process?

**Example**: I have 10 steps, but on a specific scenario, i want to start on the step 5, and stop the tour on the step 8, doing when i initialize the tour, on the array of steps. 

Is that any possibility to do that? I'm asking, because i try to stop on the step 8, but the tour continues. I know that have a way to do that on the HTML, calling the method `this.tourService.end()` but i like to know if's there any possibility to stop this action doing on the initializing process, or some property like **endOn**, since this package have a `this.tourService.startAt` function to select where is gonna start. (Maybe this property or function exist. If it is the case, sorry, i didn't see that. if's not, it might be a good idea :D)

**Comment by hakimio** [comment, marked as answer]
Hi Lucas,

You should be able to just initialize the tour with the step range you want at any point: 
```ts
this.tourService.initialize(this.steps.slice(4, 8));
```

**Reply by lucasmp2001** [reply]
Thanks for the answer! here i found another way to do that, subscribing on the event, and looking for the steps, and when come to the 8 step, the tour is finished. but Thanks!!
